packages/package-bundler/src/rollup.ts

```diff
@@ -1,70 +1,25 @@
-import alias, { RollupAliasOptions } from "@rollup/plugin-alias"
-import { babel, getBabelOutputPlugin } from "@rollup/plugin-babel"
+import { RollupAliasOptions } from "@rollup/plugin-alias"
 import commonjs from "@rollup/plugin-commonjs"
 import typescript from "@rollup/plugin-typescript"
 import { InputPluginOption, RollupOptions } from "rollup"
-import ignore from "rollup-plugin-ignore"
-import nodeExternals from "rollup-plugin-node-externals"
-import { rollupOptionsUiLibrary } from "./presets/ui-library/index.js"
-
-type Presets = "ui-library"
-
-const getPreset = (preset: Presets): RollupOptions => {
-  switch (preset) {
-    case "ui-library":
-      return rollupOptionsUiLibrary
-  }
-}
+import { presetDefault, presetUiLibrary } from "./presets/index.js"
 
 type Config = {
   input: RollupOptions["input"]
-  preset: Presets
+  plugins: InputPluginOption[]
   alias?: RollupAliasOptions
 }
 
 export const rollupConfig = (
   config: Config = {
     input: { index: "./src/index.ts" },
-    preset: "ui-library",
+    plugins: presetUiLibrary,
   }
 ): RollupOptions[] => {
-  const preset = getPreset(config.preset)
-
   // Shared config
   const sharedConfig = {
-    ...preset,
     input: config.input,
-    plugins: [
-      ...(preset.plugins as InputPluginOption[]),
-      nodeExternals({
-        devDeps: true,
-      }),
-      // Has to be the same as packages/components/tsconfig.json -> compilerOptions -> paths
-      alias(config.alias),
-      // This call to alias plugin will be additional to the above alias plugin call
-      alias({
-        entries: [
-          // i18n-react-intl package attempts to import locales from this path.
-          // When rollup attempts to import from the 'find' path, it will be
-          // redirected to import from the replacement path (same as storybook webpack config).
-          {
-            find: "__@cultureamp/i18n-react-intl/locales",
-            replacement: "locales",
-          },
-        ],
-      }),
-      // These libraries aren't used in KAIO, and require polyfills to be set up
-      // in consuming repos. Ignoring them here removes the need for extra setup in
-      // consuming repos.
-      ignore(["stream", "http", "https", "zlib"]),
-      babel({ babelHelpers: "bundled" }),
-      getBabelOutputPlugin({
-        plugins: [
-          "@babel/plugin-transform-react-pure-annotations",
-          "babel-plugin-pure-static-props",
-        ],
-      }),
-    ],
+    plugins: [...config.plugins, ...presetDefault({ alias: config.alias })],
```

Comment:
any reason not to pass whole config to preset?
```suggestion
    plugins: [...config.plugins, ...presetDefault(config)],
```

Reply:
Not technically, but I don't want to pass anything it doesn't need. Admittedly the lines are blurring here, cause if we send in all config, then we may as well pull all the config into this file again?

Reply:
My take is that it's matter of separating concerns/coupling
- you want the default preset to be in seperate file so it's more self sufficient
- also you don't want to assume from here what that file needs from the config

Reply:
also when I think about it, the default preset should be inside the ui library preset. So it allows consumer to eject from default if they need to

Reply:
Hm yeah makes sense, so there's 3 options:
- UI Library preset (inherits default preset)
- Default Preset
- None

Reply:
yes